cmd/verify_release/main.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 The Project Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// verify_release is a tool to verify a release manifest.
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"golang.org/x/mod/sumdb/note"
+)
+
+var (
+	publicKeyFile = flag.String("public_key", "", "Path to file containing the public key used to sign the manifest")
+	manifest      = flag.String("manifest", "", "Path to the signed manifest")
+)
+
+func main() {
+	flag.Parse()
+	if err := validateFlags(); err != nil {
+		glog.Exitf("Invalid flag(s):\n%s", err)
+	}
+
+	msg, err := os.ReadFile(*manifest)
+	if err != nil {
+		glog.Exitf("failed to read manifest file: %v", err)
+	}
+
+	glog.Info("Verifying signature...")
+	body, err := verify(msg)
+	if err != nil {
+		glog.Exitf("Failed to verify signature: %v", err)
+	}
+
+	fmt.Println(string(body))
+}
+
+// verify verifies the passed Go sumdb's note
+func verify(msg []byte) ([]byte, error) {
+	k, err := os.ReadFile(*publicKeyFile)
```

Comment:
This seems slightly sketchy - e.g. I could generate a new keypair, sign a bad release file with that key and create a PR which adds that release manifest to the log *and* overwrites the pubkey file in the repo.

it'd very likely get caught before merging, and AD devices presumably will have the release key baked in so the damage is limited, but it'd be a little embarrassing if it got through...

I'd be tempted to have this pub key in the github action ENV/SECRETs (but totally fine to have it checked into a file for others to access too though)

Reply:
I see your concern. Let's add the option to have the pub key in the environment then.
Nevertheless, the current leaf_validator would not allow such a PR with `PR mixes log additions and non-log changes, please split them up`.

Reply:
Yeah, this sort of this is exactly why I added that check (another attack would be to overwrite the repo's `.github/workflow/...` configs in the same PR as a "bad" queue.

Defence in depth is good.

Reply:
Fixed in 0e966eafb054718efaf114e7acf6437a48a4ceb7